### ISSUE TITLE
Feature request: format GROQ queries in JS files

### ISSUE THREAD

**Issue by coreyward** [alevroub/prettier-plugin-groq#6]
All of my GROQ is inside of files with an extension like `.js` or `.jsx` or `.ts`, and this plugin doesn't work on them, even when they are tagged as GROQ (example below) and recognized as GROQ by VSCode. It would be great to add formatting in these contexts. 

E.g.: 
```groq
const response = client.fetch(
  `//groq
    *[_type == "featureRequest]
  `
)
```

**Comment by alejandrovrojas** [comment]
Not sure how to go about nested scopes – though the VS Code integration works a little differently meaning the plugin would have to be bundled with [the official extension](https://github.com/sanity-io/vscode-sanity).

**Comment by coreyward** [comment]
@rexxars Would love your input on the above!

**Comment by rexxars** [comment]
It looks like prettier isn't really built with this in mind - the formatting of graphql, markdown etc inside of JS files seems to be [special-cased](https://github.com/prettier/prettier/blob/1ecae48ae859254cc1e3b33bb68f454b77cefc4a/src/language-js/embed.js#L241-L268) in the JS handler.

This means we'd have to PR the official prettier repo to add support for this, and I don't have high hopes that it would go through, unfortunately.

Another approach would be for the VS Code plugin to pick up template literals on save, run them through prettier and inject them back into the source, but that doesn't seem trivial to implement - we'd need to look into the formatting APIs, re-align the query with the indentation level of the code and similar, which is what prettier is supposed to take care of for us.

I like the idea, but don't feel like I can contribute with the amount of effort needed for this right now.

**Comment by coreyward** [comment]
@rexxars Thanks for looking into it. Do you happen to know how this works with GraphQL tagged template literals in Gatsby? Running `prettier --support-info` in a repo where this works prints the following info mentioning GraphQL, if helpful:

```json
{
  "aceMode": "text",
  "extensions": [".graphql", ".gql", ".graphqls"],
  "linguistLanguageId": 139,
  "name": "GraphQL",
  "parsers": ["graphql"],
  "since": "1.5.0",
  "tmScope": "source.graphql",
  "type": "data",
  "vscodeLanguageIds": ["graphql"]
}
```

To be clear about what I mean, when I run my formatter (configured as Prettier) on a JS(X) file like this, the GraphQL portion is formatted with a GraphQL-aware formatter:

```js
import { graphql } from "gatsby"

export const query = graphql`
  {
    foo(id: 42) {
      bar
    }
  }
`
```

**Comment by rexxars** [comment]
> Do you happen to know how this works with GraphQL tagged template literals in Gatsby?

Yeah, as I mentioned above there is [special handling](https://github.com/prettier/prettier/blob/1ecae48ae859254cc1e3b33bb68f454b77cefc4a/src/language-js/embed.js#L241-L268) of graphql, markdown and similar if using a set of predefined tagged template literals (`graphql`, `gql` etc) inside of the javascript formatting logic of prettier. For this reason it's not easy to extend it to other template literals, since there doesn't seem to be any "extension points"

**Comment by coreyward** [comment]
Is this a pattern that would work for supporting GROQ in JS? https://github.com/prettier/prettier/pull/12707

**Comment by xndyz** [comment]
This would be insanely useful given groq queries are usually big and Sanity is often used with template literals.
